src/Compilers/CSharp/Portable/Symbols/ExtraAnnotations.cs

```diff
@@ -190,7 +196,7 @@ internal static class ParameterAnnotationsExtensions
     {
         // For EnsuresNotNull, you should set NotNullWhenTrue and NotNullWhenFalse
         internal static AttributeAnnotations With(this AttributeAnnotations value,
-            bool notNullWhenTrue, bool notNullWhenFalse)
+            bool notNullWhenTrue, bool notNullWhenFalse, bool ensuresTrue, bool ensuresFalse)
```

Comment:
There are no callers of this extension method that pass a non-default `value`. Consider removing this method and having the two callers do the `|` operations explicitly.